### PR TITLE
ignore errors if ignore_saved_search flag is true

### PR DESCRIPTION
**Pull request by vishalxebia**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Versioning

- [ ] Bumped the version check in `manifest.json`
- [ ] Updated `CHANGELOG.md`

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: splunk_event/test_splunk_event.py [hunk 1]
```diff
@@ -993,6 +993,46 @@ def _mocked_saved_searches(*args, **kwargs):
...
+        self.maxDiff = None
+
+        config = {
+            'init_config': {},
+            'instances': [
+                {
+                    'url': 'http://localhost:13001',
+                    'username': "admin",
+                    'password': "admin",
+                    'ignore_saved_search_errors': 'true',
```
Comment (whazenberg) [inline splunk_event/test_splunk_event.py]: does value 'true' without the quotes also work? Maybe using the native boolean value may be cleaner than this string variant.

File: splunk_topology/check.py [hunk 2]
```diff
@@ -125,7 +125,9 @@ def check(self, instance):
         except Exception as e:
             self._clear_topology(instance_key, clear_in_snapshot=True)
             self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=instance.tags, message=str(e))
-            raise CheckException("Splunk topology failed with message: %s" % e), None, sys.exc_info()[2]
+            if not instance.splunk_ignore_saved_search_errors:
+                raise CheckException("Splunk topology failed with message: %s" % e), None, sys.exc_info()[2]
+            self.log.warning("Splunk topology exception occured and ignored as ignore_saved_search_errors flag is true.")
```
Comment (whazenberg) [inline splunk_topology/check.py]: maybe log the actual error message when the flag is set to true.

File: splunk_topology/check.py [hunk 3]
```diff
@@ -139,8 +141,10 @@ def _dispatch_and_await_search(self, instance, saved_searches):
                     self._finalize_sid(instance, sid, saved_search)
                     self.update_persistent_status(instance.instance_config.base_url, saved_search.name, sid, 'remove')
             except FinalizeException as e:
-                self.log.error("Got an error %s while finalizing the saved search %s" % (e.message, saved_search.name))
-                raise e
+                if not instance.splunk_ignore_saved_search_errors:
+                    self.log.error("Got an error %s while finalizing the saved search %s" % (e.message, saved_search.name))
+                    raise e
+                self.log.warning("Ignoring finalize exception as ignore_saved_search_errors flag is true.")
```
Comment (whazenberg) [inline splunk_topology/check.py:148]: maybe log the actual error when the flag is set to true

Comment (whazenberg) [reply splunk_topology/check.py:148]: maybe include a piece of specific information in where the error occurred? This message is very similar as to previous message. This message may be fired on dispatching while the other might be fired during finalization.

File: splunk_topology/check.py (hunk 2 above)
Comment (whazenberg) [reply splunk_topology/check.py]: maybe include a piece of specific information in where the error occurred? This message is very similar as to next message. This message may be fired on dispatching while the other might be fired during finalization.

File: splunk_topology/test_splunk_topology.py [hunk 4]
```diff
@@ -207,7 +208,7 @@ def _mocked_finalize_sid_exception(*args, **kwargs):
         except CheckException:
             thrown = True
 
-        self.assertTrue(thrown)
+        self.assertFalse(thrown)
```
Comment (whazenberg) [inline splunk_topology/test_splunk_topology.py]: do we now have tests that at least cover both cases, i.e. a case where `ignore_saved_search_errors` is set to `true` and a case where it set to `false`?

**Review by whazenberg — APPROVED** [review]

File: splunk_event/test_splunk_event.py (hunk 1 above)
Comment (vishalxebia) [reply splunk_event/test_splunk_event.py]: changed to boolean value.

File: splunk_topology/check.py (hunk 3 above)
Comment (vishalxebia) [reply splunk_topology/check.py:148]: showing the actual error as well.

File: splunk_topology/test_splunk_topology.py (hunk 4 above)
Comment (vishalxebia) [reply splunk_topology/test_splunk_topology.py]: added a new case for true.

**Review by whazenberg — APPROVED** [review]

**Pull request merged**